simple/imm_data.c

```diff
@@ -105,20 +121,20 @@ static int alloc_ep_res(struct fi_info *fi)
 	cq_attr.size = rx_depth;
 	ret = fi_cq_open(dom, &cq_attr, &rcq, NULL);
 	if (ret) {
-		printf("fi_cq_open() send comp %s\n", fi_strerror(-ret));
+		FI_PRINTERR("fi_cq_open: recv completion", ret);
```

Comment:
I hit into the same issue trying to use FI_PRINTERR.  If we start adding variable arguments, it isn't much better than just calling fprintf directly.

One possibility is to have do something like this:

#define FI_PRINTERR(ret) \
   fprintf(stderr, "%s():%d, %d (%s)\n", __func__, __line__, ret, fi_strerror)

Also, I would consider renaming it to FT_PRINTERR, and avoid using the "FI" prefix, which should be reserved for libfabric itself.